serverless/serverless
# AWS Go support

## What did you implement:

Closes #4507 

[AWS Lambda supports Go now](https://aws.amazon.com/about-aws/whats-new/2018/01/aws-lambda-supports-go/), serverless should support it too.

I will leave the local invoke to another PR. I think I know how to do it.

Note I called the plugin `aws-go` instead of `aws-golang` to be consistent with AWS library name: https://github.com/aws/aws-lambda-go

## How did you implement it:

I leveraged @mthenw's template https://github.com/serverless/serverless-golang which did all the work for me, and then followed @horike37's step by step instructions in the above issue.

There are 2 templates:

1. aws-go - vanilla `go get`
2. aws-go-dep - uses `go dep` 

Each template contains two functions: `hello/main.go` and `world/main.go`

## How can we verify it:

Run `npm test` to make sure all tests pass.

Run the following commands to make sure you can create a go lambda using both templates:

    sls create -t aws-go -p myservice

Or

    sls create -t aws-go-dep -p myservice

Run the following to build and deploy:

    cd aws-go
    ./build.sh
    sls deploy
    sls invoke -f hello
    sls invoke -f world

Run integration test:

    integration-test aws-go 'cd /go/src/app && ./build.sh'
    integration-test aws-go-dep 'cd /go/src/app && ./build.sh'

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
